### PR TITLE
Suggest `make clean` in Sanitizers tutorial

### PR DESCRIPTION
**Pull request by nskh**
Adds a warning to run `make clean` after enabling sanitizers in Makefile to the Sanitizers tutorial

### CODE REVIEW

File: docs/setup_asan.md [hunk 1]
```diff
@@ -16,6 +16,10 @@ If you're interested, here is more information about the [Address Sanitizer](htt
 ## Quick Start
 To enable the address sanitizer, add these compiler flags to add to your `Makefile`.
 
+<div class="primer-spec-callout warning" markdown="1">
+**Note:** If you enable sanitizers in your `Makefile` and change nothing else in your code, you will need to do a `make clean` to recompile with the sanitizers on.
```
[the remark this answers is not in the document]
Comment (awdeorio) [reply docs/setup_asan.md:20]: I'll also suggest moving this callout to the end of the subsection.

**Review by awdeorio — CHANGES_REQUESTED** [review]
Great idea!  I left a couple of suggestions.

File: docs/setup_asan.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jamesjuett) [reply docs/setup_asan.md:20]: My 2 cents... the pitfall wording may be unclear to students as written. i.e. Is it a pitfall to do this? Or to forget to do it?

[the remark this answers is not in the document]
Comment (nskh) [reply docs/setup_asan.md:20]: I agree with James; we could say 
> To ensure your compiled code runs with sanitizers, remove and rebuild your executables after changing the `Makefile`. You can do this with `make clean`, for example.

It's a little more heavy-handed but is probably clearer for 280 students who may not know where to find, remove, and recompile executables.

[the remark this answers is not in the document]
Comment (awdeorio) [reply docs/setup_asan.md:20]: How about we go back to mostly the original text, including the label Pitfall to be consistent with other yellow callouts and including a `$ make clean` in a code block?  I'll revise my suggestion above.

[the remark this answers is not in the document]
Comment (nskh) [reply docs/setup_asan.md:20]: I like that a lot!

**Review by awdeorio — APPROVED** [review]
LGTM

**Pull request merged**